src/App.js

```diff
@@ -9,17 +9,69 @@ import ReactDOM from 'react-dom';
 import Slider from './Components/Slider';
 import Table from './Components/Table';
 import mandalartImage from '../src/assets/images/mandalart_otani.jpeg';
-import { useState } from 'react';
+import { useEffect, useState } from 'react';
 
 function Portal({ children }) {
   return ReactDOM.createPortal(children, document.getElementById('modal'));
 }
 
+const MAIN = [...new Array(9)].fill('');
+const SUB = [...new Array(9)].map(() => [...new Array(9)].fill(''));
+
 function App() {
-  const [mainInput, setMainInput] = useState([...new Array(9)].fill(''));
+  const [mainInput, setMainInput] = useState(MAIN);
+  const [subInput, setSubInput] = useState(SUB);
   const [name, setName] = useState('');
   const [isModalOpen, setIsModalOpen] = useState(false);
 
+  const setInputValue = (tableKey, index, value) => {
+    const id = `box-${tableKey}-${index}`;
+    const element = document.getElementById(id);
+    if (element) {
+      element.textContent = value;
+    }
+  };
+
+  // initialize main/sub input with localStorage
+  useEffect(() => {
+    const m = localStorage.getItem('mainInput');
+    const s = localStorage.getItem('subInput');
+    if (m) {
+      setMainInput(JSON.parse(m));
+      for (var j = 0; j < 9; j++) {
+        setInputValue('main', j, JSON.parse(m)[j]);
+      }
+    }
+    if (s) {
+      setSubInput(JSON.parse(s));
+      for (var i = 0; i < 9; i++) {
+        for (var j = 0; j < 9; j++) {
+          setInputValue(i + 1, j, JSON.parse(s)[i][j]);
+        }
+      }
+    }
+  }, []);
+
+  const handleReset = () => {
+    setName('');
+    setIsModalOpen(false);
+
+    // localstoage
+    localStorage.setItem('mainInput', JSON.stringify(MAIN));
+    localStorage.setItem('subInput', JSON.stringify(SUB));
+
+    // input values
+    setMainInput(MAIN);
+    setSubInput(SUB);
+    for (var i = 0; i < 9; i++) {
+      for (var j = 0; j < 9; j++) {
+        setInputValue(i + 1, j, '');
+      }
+      setInputValue('main', i, '');
+    }
+    alert('초기화 되었습니다.');
+  };
```

Comment:
I think this is already great, but what do you think about putting one more hurdle before initialization? 🤩 
Just in case that button is pressed against by people's intension.
things like..

```js
  const handleReset = () => {
    if (window.confirm('초기화 하시겠습니까?') {
       setName('')
       ...
       alert('초기화 되었습니다.')
    }
  }
```